src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -612,19 +791,105 @@ private SnapshotSummaryModel getSnapshotSummary(UUID snapshotId) {
         snapshotId, snapshotService::retrieveSnapshotSummary);
   }
 
+  @VisibleForTesting
+  DRSObject mergeDRSObjects(List<DRSObject> drsObjects) {
+    DRSObject drsObject =
+        new DRSObject()
+            // Extract singleton values
+            .id(extractUniqueDrsObjectValue(drsObjects, DRSObject::getId))
+            .name(extractUniqueDrsObjectValue(drsObjects, DRSObject::getName))
+            .description(extractUniqueDrsObjectValue(drsObjects, DRSObject::getDescription))
+            .size(extractUniqueDrsObjectValue(drsObjects, DRSObject::getSize))
+            .selfUri(extractUniqueDrsObjectValue(drsObjects, DRSObject::getSelfUri))
+            .mimeType(extractUniqueDrsObjectValue(drsObjects, DRSObject::getMimeType))
+            .version(extractUniqueDrsObjectValue(drsObjects, DRSObject::getVersion))
+            // Get the earliest creation date
+            .createdTime(
+                drsObjects.stream()
```

Comment:
Food for thought, it might be easier to do more targeted and complete unit testing of this method if these stream operations were broken out into their own methods and then tested individually.  That could also sidestep the need for the comments and make the merge more readable. Example:

```
…
  .createdTime(getEarliestCreatedTime(drsObjects))
…

@VisibleForTesting
String getEarliestCreatedTime(List<DRSObject> drsObjects) {
  return drsObjects.stream()
                    .map(DRSObject::getCreatedTime)
                    .map(Instant::parse)
                    .min(Comparator.naturalOrder())
                    .map(Instant::toString)
                    .orElse("");
}
```